pkg/trace/metrics/timing/timing.go

```diff
@@ -0,0 +1,104 @@
+package timing
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/atomic"
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+)
+
+// Set represents a set of metrics that can be used for timing. Use NewSet
+// to create a set. It is safe for concurrent use.
+type Set struct {
+	c   map[string]*counter // maps names to their aggregates
+	ctx context.Context     // used for cancelling when auto-reporting
+}
+
+// NewSet returns a new Set containing the given metric names. The context is optional
+// and can be used as cancellation for auto-reporting sets.
+func NewSet(ctx context.Context, names ...string) *Set {
+	set := Set{
+		c:   make(map[string]*counter, len(names)),
+		ctx: ctx,
+	}
+	for _, name := range names {
+		set.c[name] = newCounter(name)
+	}
+	return &set
+}
+
+// Autoreport enables autoreporting of the Set at the given interval.
+func (s *Set) Autoreport(interval time.Duration) {
+	go func() {
+		tick := time.NewTicker(interval)
+		defer tick.Stop()
+		for {
+			select {
+			case <-tick.C:
+				s.Report()
+			case <-s.ctx.Done():
+				return
+			}
+		}
+	}()
+}
+
+// Measure measures the time passed since, for the given name. It aggregates on each
+// subsequent call until Report is called.
+func (s *Set) Measure(name string, since time.Time) {
+	ms := float64(time.Since(since) / time.Millisecond)
+	c, ok := s.c[name]
+	if !ok {
+		panic(fmt.Sprintf("Set: key does not exist: %s", name))
+	}
+	c.add(ms)
+}
+
+// Report reports all of the Set's metrics to the statsd client.
+func (s *Set) Report() {
+	for _, c := range s.c {
+		c.flush()
+	}
+}
+
+type counter struct {
+	// name specifies the name of this counter
+	name string
+
+	// mu synchronizes calls between add and flush, ensuring
+	// that adding doesn't happen while flushing.
+	mu sync.RWMutex
+
+	count, max, sum *atomic.Float64
```

Comment:
What about `min`?

Reply:
We could add it, but I didn't see any use for it. It would just show us the fastest, which isn't interesting unless I missed something. It could always be added later if a need arises.

Reply:
Might be good to add just for consistency with how the statsd timing works.

Reply:
> we don't have a p95.

That would be nice as well 😛 

Yeah, in this case we really only care about average. Meh, could keep it simple now and add later if we decide we need it.